lib/http.c

```diff
@@ -2046,8 +2046,13 @@ static CURLcode set_reader(struct Curl_easy *data, Curl_HttpReq httpreq)
       else
         result = Curl_creader_set_null(data);
     }
-    else { /* we read the bytes from the callback */
-      result = Curl_creader_set_fread(data, postsize);
+    else {
+      /* we read the bytes from the callback. In case "chunked" encoding
+       * is forced by the application, we disregard `postsize`. This is
+       * a backward compatibility decision to earlier versions where
+       * chunking disregarded this. See issue #13229. */
+      bool chunked = !!Curl_checkheaders(data, STRCONST("Transfer-Encoding"));
```

Comment:
I think should instead use `data->req.upload_chunky` ?

Reply:
No, that does not work. That flag is set after the installed reader is checked for known length.

Reply:
Aha! But then maybe the checks should at least be done the same way? 
~~~c
  bool chunked = FALSE;
  char *ptr = Curl_checkheaders(data, STRCONST("Transfer-Encoding"));
  if(ptr) {
    /* Some kind of TE is requested, check if 'chunked' is chosen */
    chunked =
      Curl_compareheader(ptr,
                         STRCONST("Transfer-Encoding:"), STRCONST("chunked"));
  }

Reply:
Right. It works for Git's case because we are setting `chunked`, but I guess some other caller setting `Transfer-Encoding: gzip` or something would confuse things. My understanding is that you can pass multiple items there (e.g, `Transfer-Encoding: gzip, chunked`), and I don't think that would match correctly here. But AFAICT none of that is new (and is certainly not something Git does).

Reply:
Just added the check as suggested by @bagder.

We have a bit backward compatibility weirdness here. Ideally, we would either auto-reset post sizes between transfers OR require all applications to set it. But that would certainly also break some apps.